crux-core/src/crux/query.clj

```diff
@@ -1087,13 +1087,17 @@
                            q)
                          identity
                          (fn [q]
-                           (let [q (normalize-query (s/assert ::query q))]
-                             (->ConformedQuery q (s/conform ::query q)))))]
+                           (if-let [s-error (s/explain-data ::query q)]
```

Comment:
couple of things with the general pattern here:

* could you use `s/valid?` for the initial check, so that the initial check's quick for the normal case, where the query's valid? don't mind it using slower functions once we know the query's invalid, but trying to keep the fast-path as fast as possible.
* if memory serves there's an easier way to get the error data out than the dance you're having to do here with `with-out-str` - could you have a look through the other functions spec offers?
* I try to use a pattern of 
```
(when invalid? (throw ...))

(process-valid-query ...)
```
rather than
```
(if invalid? 
  (throw ...)
  (process-valid-query ...))
```
to avoid the extra nesting for the happy path - more important if there are more validity checks, obviously